src/tests/miri.rs

```diff
@@ -0,0 +1,19 @@
+// Simple tests using MIRI
+
+use crate::{memchr, memchr2, memchr3, memrchr, memrchr2, memrchr3};
+
+#[test]
+fn test_with_miri() {
+    assert_eq!(memchr(b'a', b"abcda"), Some(0));
+    assert_eq!(memchr(b'z', b"abcda"), None);
+    assert_eq!(memchr2(b'a', b'z', b"abcda"), Some(0));
+    assert_eq!(memchr2(b'z', b'y', b"abcda"), None);
+    assert_eq!(memchr3(b'a', b'z', b'b', b"abcda"), Some(0));
+    assert_eq!(memchr3(b'z', b'y', b'x', b"abcda"), None);
+    assert_eq!(memrchr(b'a', b"abcda"), Some(4));
+    assert_eq!(memrchr(b'z', b"abcda"), None);
+    assert_eq!(memrchr2(b'a', b'z', b"abcda"), Some(4));
+    assert_eq!(memrchr2(b'z', b'y', b"abcda"), None);
+    assert_eq!(memrchr3(b'a', b'z', b'b', b"abcda"), Some(4));
+    assert_eq!(memrchr3(b'z', b'y', b'x', b"abcda"), None);
+}
```

Comment:
Out of curiosity, why are these needed? Why can't miri just use the normal test suite?

(Sorry if this is a stupid question, I haven't actually used miri yet!)

Reply:
MIRI is magnitudes slower (100x, maybe even 1000x/10000x) than binaries built in debug mode
I tried running the test suite, the first test would show up as complete relatively quickly but no other test would complete for another few minutes
I assume there are timeout limits on GitHub actions, hence the short test, I suppose that the existing test suite could work just fine, I didn't really have the time to stick around :)

Reply:
Ah gotya. Yeah the tests check all the possible different offsets and alignments. This gets quite big because of the 256 bit vectors in the AVX version. It's probably possible to merge the test suites and just run fewer iterations when miri is enabled. I'll look into that when I get a chance!